File: Demo/TimerDemoObjC/TimerDemo7/ViewController.m
//
//  ViewController.m
//  TimerDemo7
//
//  Created by frank on 18/09/2016.
//  Copyright © 2016 frank. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@property (nonatomic, weak) IBOutlet UILabel* counterLabel;
@property (nonatomic, assign) int counterValue;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    [self updateCounterWithValue:0];
    [NSTimer scheduledTimerWithTimeInterval:.5 target:self selector:@selector(timerDidFire:) userInfo:nil repeats:YES];
}


- (void)timerDidFire:(NSTimer*)timer
{
    [self updateCounterWithValue:self.counterValue + 1];
}

- (void)updateCounterWithValue:(int)newValue
{
    self.counterValue = newValue;
    self.counterLabel.text = [NSString stringWithFormat:@"%d", newValue];
}

@end

File: Demo/TimerDemoSwift/TimerDemoSwift/ViewController.swift
//
//  ViewController.swift
//  TimerDemoSwift
//
//  Created by Frank on 19/09/16.
//  Copyright © 2016 Frank Lefebvre. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    var counterValue = 0
    @IBOutlet var counterLabel: UILabel!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        NSTimer.scheduledTimerWithTimeInterval(0.5, target: self, selector: #selector(ViewController.timerDidFire), userInfo: nil, repeats: true)
        updateCounterWithValue(0)
    }

    func timerDidFire(timer: NSTimer) {
        updateCounterWithValue(counterValue + 1)
    }
    
    func updateCounterWithValue(newValue: Int) {
        FrenchKitUpdateCounter(newValue)
        counterValue = newValue
        counterLabel.text = "\(newValue)"
    }
}

